projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/SentenceService.kt

```diff
@@ -67,9 +69,57 @@ class SentenceService(
             requirements = requirementRepository.getRequirements(id, eventNumber)
                 .map { it.toRequirement() },
             courtDocuments = documentRepository.getCourtDocuments(id, eventNumber).map { it.toCourtDocument() },
-            disposal?.id?.let { getUnpaidWorkTime(it) }
+            disposal?.id?.let { getUnpaidWorkTime(it) },
+            licenceConditions = disposal?.let {
+                licenceConditionRepository.findAllByDisposalId(disposal.id).map {
+                    it.toLicenceCondition()
+                }
+            } ?: emptyList(),
+        )
+
+    fun EntityLicenceCondition.toLicenceCondition() =
+        LicenceCondition(
+            mainCategory.description,
+            subCategory?.description,
+            imposedReleasedDate,
+            actualStartDate,
+            populateLicenceConditionNotes(notes)
         )
 
+    fun populateLicenceConditionNotes(notes: String?): List<LicenceConditionNote> {
+        val noteLength = 1500
+
+        notes?.let {
+            val splitParam = "---------------------------------------------------------" + System.lineSeparator()
+            return notes.split(splitParam).map {
+                val noteSplitList = it.split(System.lineSeparator())
+                val note = noteSplitList.drop(1).joinToString(System.lineSeparator())
+
+                LicenceConditionNote(
+                    noteSplitList.first(),
+                    note.chunked(noteLength)[0],
+                    note.let { n ->
+                        when {
+                            n.length > noteLength -> true
+                            else -> false
+                        }
+                    }
+                )
+            }
+
+        }
```

Comment:
Not 100% sure what this is trying to do?  Is it to move the "Comment added by ..." line out?

Reply:
yes